### ISSUE TITLE
code flow with PKCE: renewal removes id_token and isAuthenticated becomes false

### ISSUE THREAD

**Issue by dschmidt** [perarnborg/vuex-oidc#91]
Hey,

I'm not completely sure this qualifies as a bug, but I hope you can guide me on how to deal with this correctly.

I'm using the vuex-oidc-example repo with the following settings with response type "code" and scope "openid profile offline_access". This correctly uses the PKCE flow and retrieves a refresh token. When I click "Reauthenticate silently" the refresh_token is correctly used to retrieve a new access_token. So far so good.
When the new access_token is retrieved, the response does not contain an id_token and id_token in store/localStorage gets overriden with an empty value. That in turn makes `isAuthenticated` false and the example page shows me as not logged in anymore. 

1) To me it seems fishy the id_token is erased
2) even if that's worked around by merging the old id_token with the new response. what should determine the logged in state? how should expiry be handled? 

in a lot of cases it's possibly more relevant whether there still is a valid access_token than if there's an id_token. I can possibly just define my own getters, but should the example handle that too? 

Thanks in advance.

PS: ftr I'm using kopano konnect as IdP.

**Comment by dschmidt** [comment]
FTR this is what the content of the localStorage looks like after refreshing. It's basically the same as before refreshing. Just - as I said - the id_token is removed.

```json
{
  "session_state": "....",
  "access_token": "....",
  "refresh_token": "...",
  "token_type": "Bearer",
  "scope": "openid profile offline_access",
  "profile": {
    "sub": "mysub@konnect",
    "sid": "...",
    "family_name": "Schmidt",
    "given_name": "Dominik",
    "name": "Schmidt, Dominik",
    "preferred_username": "dschmidt"
  },
  "expires_at": 1586886864
}
```

**Comment by perarnborg** [comment]
Thank you for posting this!

I agree that it does not feel right that we erase the id_token in this case. I beleive that the provider implementations that I have looked at do not have this issue. I need to look at this more closely, I'll get back to you.

The issue of what should determine isAuthenticated has been up before, and I thunk it could be useful to allow implementors to define this with a setting.

**Comment by dschmidt** [comment]
Thanks for the quick response!

I had found https://github.com/perarnborg/vuex-oidc/issues/20 before, but I wasn't sure how it did (or didn't) apply to my specific case - yes, a setting sounds good.

Tbh I'm just looking into oidc and it's really hard for me to tell what part of the system is not behaving well when something is not working. I had a friend test the example app with his keycloak installation and he did not suffer from the same issue apparently. So indeed it seems to be provider specific.

**Comment by dschmidt** [comment]
I've sent 3 pull requests
- #93 adds a setting to switch to be able to switch to access_token for isAuthenticated determination
- #92 adds refresh tokens to the store
- https://github.com/perarnborg/vuex-oidc-example/pull/20 shows all available tokens and expiration times in the example app

With those PRs together I can apparently use the example app with code/PKCE flow, see protected pages and (at least manual) access_token renewals work fine for me.

**Comment by perarnborg** [comment]
Perfect! See comments on #93. Good job!

**Comment by perarnborg** [comment]
Published in v3.5.0
